### ISSUE TITLE
Reading multiple shards in parallel, while treating each shard as a sequence of data (e.g. video/audio) 

### ISSUE THREAD

**Issue by cxlcl** [tmbdev/webdataset#55]
Hi, 

This is a very nice project and thanks for sharing it! 

I'm wondering if there is a way to read multiple shards in parallel, while treating each shard as a sequence of data, such as video or audio.  For example, assuming there are two .tar files each including the frames of a video. The code snippet I used for loading them in parallel is:

```
ds = wds.Dataset([
    "video-0.tar", 
    "video-1.tar"]).decode()

batch_size=2
dataset = (
        wds.dataset.Processor(ds, wds.map, my_sample_decoder) 
        .batched(batch_size))

dataloader = torch.utils.data.DataLoader(dataset, num_workers=2, batch_size=None,)

for sample in dataloader:
    # process each batched sample#
```

The sequence of frames loaded in each iteration are: `[frame_0_in_video_0, frame_1_in_video_0]`, then `[frame_0_in video_1, frame_1_in_video_1]`, then `[frame_2_in_video_0, frame_3_in_video_0]` and so on, while the behavior I was trying to implement is: `[frame_0_in_video_0, frame_0_in_video_1]`, then `[frame_1_in_video_0, frame_1_in_video_1]` and so on.

**Comment by tmbdev** [comment]
I'm assuming that video-0.tar and video-1.tar actually contain individual image files corresponding to the frames of the videos.

The simple answer is that you can just create a custom IterableDataset. This looks roughly like the following:

```
class MergeVideos(IterableDataset, Composable, Shorthands):
    def __init__(self, video0, video1, ...):
        self.videos = [video0, video1]
    def __iter__(self):
        video0, video1 = self.videos
        ds0 = WebDataset(video0, ...).decode(...)
        ds1 = WebDataset(video1, ...).decode(...)
        for sample0, sample1 in zip(ds0, ds1):
            result = {"v0.jpg": sample0["jpg"], "v1.jpg": sample1["jpg"]}
            yield result

for sample in MergeVideos(video0, video1).shuffle(...).decode(...):
    ...
```

However, it's generally better not to do this on the fly; you're better off merging the two videos ahead of time and writing the merged dataset out into separate shards. That simplifies debugging, permits better shuffling, and speeds up I/O.
